### DISCUSSION TITLE
Figure supplement and numbering styling

### DISCUSSION THREAD

**Discussion by CDucloux** [typst/typst#3871] [Q&A]
I know that it is possible to style the entire caption of a `Figure`, however, **I'm specifically interested in styling the supplement AND the numbering part, using bold and color (orange)**.

```typst
#set figure.caption(
  separator: [ --- ]
)

#figure(
  image("cat.jpg", width: 40%),
  caption: [A cute cat],
  supplement: [#text("Figure", fill: orange)],
) <fig>
```

Here is my current version : I've also been able to change the color of the separator, but unable, so far, to change the number color, which looks kinda weird.

<img width="265" alt="example" src="https://github.com/typst/typst/assets/122461892/2d3628b4-8062-4b21-83ad-b98a78bd3368">

**Comment by xkevio** [comment]
```typ
#set figure.caption(separator: [ --- ])
#show figure.caption: c => [
  #text(fill: orange, weight: "bold")[
    #c.supplement #c.counter.display(c.numbering)
  ]
  #c.separator#c.body
]

#figure(rect(), caption: "My great caption.")
```
yields

![image](https://github.com/typst/typst/assets/13004777/decc52fe-8d01-4a16-a521-f8e01a9f8944)

**Reply by CDucloux** [reply]
Great. Thanks for the help!

The only issue is that now I would also like the cross-reference label to be entirely orange, as well as inside the list of figures.

```typst
#set ref(supplement: it => {
  if it.func() == figure {
    text("Figure", fill: orange_edf)
    //it.caption.numbering
  } else {
    "TODO."
  }
})

@fi shows a rectangle.

#set figure.caption(
  separator: [ --- ]
)

#show figure.caption: c => [
  #v(0.25cm)
  #text(fill: orange_edf, weight: "bold")[
    #c.supplement.text.replace(".", "ure") #c.counter.display(c.numbering)
  ]
  #c.separator#c.body
]

#figure(
  rect(),
  caption: [My great caption],
) <fi>
```

I've been able to partially style the cross-reference as you can see here, but not the list of figures.

<img width="322" alt="example" src="https://github.com/typst/typst/assets/122461892/109e4ba6-af26-4dd1-a3af-15379c6fc846">

*Note that my language is set to French, which explains why Figure is displayed as Fig. in the list of figures.*

<img width="771" alt="listoffigures" src="https://github.com/typst/typst/assets/122461892/05479e01-664d-47f8-950a-778f665e3d7f">
